File: Teste_VegaTech/ClientApp/src/components/Material.js
import React, { Component } from 'react';
import { Grid, TextField, Button } from "@material-ui/core";

export class Material extends Component {
  static displayName = Material.name;

    // The app should be able to automatically get the informations of "CreatedAt, CreatedBy, UpdatedAt and UpdatedBy" to insert on the database. This is not implemented here, however.
  render() {
    return (
        <form noValidate autoComplete="off">
            <Grid container>
                <Grid item xs={6}>
                    <TextField
                        id="standard-basic"
                        label="Supplier's ID"
                        name="IDSupplier"
                    />
                    <TextField
                        label="Code"
                        name="Code"
                    />
                    <TextField
                        label="Name"
                        name="Name"
                    />
                    <TextField
                        label="Description"
                        name="Description"
                    />
                    <TextField
                        label="Fiscal Code"
                        name="FiscalCode"
                    />
                    <TextField
                        label="Specie"
                        name="Specie"
                    />
                    <div>
                        <Button
                            variant="contained"
                            color="primary"
                            type="submit"
                            name="AddMaterial"
                        >
                            Add new Material
                      </Button>
                    </div>
                </Grid>
            </Grid>
        </form>
    );
  }

}

File: Teste_VegaTech/Models/Material.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Teste_VegaTech.Models
{
    public class Material
    {
        public int Id { get; set; }
        public int IdSupplier { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string FiscalCode { get; set; }
        public string Specie { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public string UpdatedBy { get; set; }
    }
}
